httpserver/bin/static_file_server.dart

```diff
@@ -19,7 +19,8 @@ void main() {
     staticFiles.serveFile(new File(indexUri.toFilePath()), request);
   };
 
-  HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4048).then((server) {
-    server.listen(staticFiles.serveRequest);
-  });
-}
\ No newline at end of file
+  var server =
+      await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4048);
+  server.forEach(staticFiles.serveRequest);
+  print('Listening on port 4048');
```

Comment:
Move the print above the server.forEach and `await` the forEach call. It might never complete, but it keeps the "async-style" behavior through the method instead of dropping a future on the floor.